paparazzi/src/main/java/app/cash/paparazzi/internal/ImageUtils.kt

```diff
@@ -312,6 +313,40 @@ internal object ImageUtils {
     }
   }
 
+  fun smallestDiffRect(firstImage: BufferedImage, secondImage: BufferedImage): Rectangle? {
+    var rect: Rectangle? = null
+    val width = max(firstImage.width, secondImage.width)
+    val height = max(firstImage.height, secondImage.height)
+
+    for (y in 0 until height) {
+      for (x in 0 until width) {
+        val firstRgb = if (x < firstImage.width && y < firstImage.height) firstImage.getRGB(x, y) else 0
+        val secondRgb = if (x < secondImage.width && y < secondImage.height) secondImage.getRGB(x, y) else 0
+        if (firstRgb != secondRgb) {
+          rect = if (rect == null) {
+            Rectangle(x, y, 1, 1)
+          } else {
+            rect.union(Rectangle(x, y, 1, 1))
+          }
+        }
+      }
+    }
+
+    return rect
+  }
+
+  fun BufferedImage.resize(targetWidth: Int, targetHeight: Int): BufferedImage {
+    if (width == targetWidth && height == targetHeight) {
```

Comment:
I don’t like this optimization because the subject is mutable. In the actually-resizing case I’m safe to further manipulate the result as much as I like; in the not-resizing case I can’t.

Maybe just don’t ?

Reply:
(And if the 1:1 case is common, and worthy of optimization, make that the caller’s problem?)